documents/coredocument.go

```diff
@@ -469,6 +463,8 @@ func (cd *CoreDocument) documentTree(docType string, mutable bool) (tree *proofs
 	if err != nil {
 		return nil, err
 	}
+
+	cd.CoreDocModified = false
```

Comment:
Why do you set it to false here? Wouldn't that lead to the flag getting unset without potentially storing it?

Reply:
Actually never mind, I think on second thought it makes sense.